### PR TITLE
Enhancement: Asset API change log documentation

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
Introduction of the Asset API change log documentation.

Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [x] This is a documentation change only

Issues:
- #349

Purpose:
Add a change log for the Asset API so game asset devs can easily determine which version they need to check for.

### CODE REVIEW

File: doc/ASSET_API.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+-----------------------------
+
+As of version 0.8.0 the Vega Strike Engine Asset API, aka Asset API, provides an interface
+for determining the versions information related to the Vega Strike Engine including:
+
+- the official version fo the Vega Strike Engine (f.e `0.8.0`)
+- the version of the Asset API implemented by the Vega Strike Engine using a simple incrementing
+  integer value.
+
+For simplicity, the Asset API Version is a simple incrementing interger value with a behavior
```
Comment (P-D-E) [inline doc/ASSET_API.rst:26]: interger -> integer

File: doc/ASSET_API.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+like that of the Android SDK version. The Asset API and the Vega Strike Engine versions have a
+loose relationship. Game Assets should use the Asset API Version as the basis for the
+determining available functionality.
```
Comment (P-D-E) [inline doc/ASSET_API.rst:29]: for the determining available functionality:
Did you mean 'for determining the available functionality' ?

File: doc/ASSET_API.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+- Renaming existing functionality (e.g someFunction to SomeFunction)
+- Removing functionality
+- Changing behavior of an existing function
+
+What kind of changes avoid changing the Engine API version?
+
+- Refactoring the Asset API in a manner that does not change its behavior.
+- Changes to code not impacting the behavior of the Asset API.
+
+Essentially, any time a change happens to the Asset API that may effect the Game Assets.
```
Comment (stephengtuggy) [inline doc/ASSET_API.rst:91]: effect -> affect (in this case)

**Review by stephengtuggy — CHANGES_REQUESTED** [review]
I have a few editorial changes to suggest. Other than that, it looks good. Glad we have this document now.

**Review by Loki1950 — APPROVED** [review]

**Review by Loki1950 — APPROVED** [review]

**Review by P-D-E — APPROVED** [review]

**Review by stephengtuggy — APPROVED** [review]
LGTM!

**Pull request merged**